bot/exts/recruitment/talentpool/_cog.py

```diff
@@ -273,53 +230,30 @@ async def _nominate_user(self, ctx: Context, user: MemberOrUser, reason: str) ->
             await ctx.send(":x: Nominating staff members, eh? Here's a cookie :cookie:")
             return
 
-        if not await self.refresh_cache():
-            await ctx.send(f":x: Failed to update the cache; can't add {user.mention}.")
-            return
-
         if len(reason) > REASON_MAX_CHARS:
             await ctx.send(f":x: The reason's length must not exceed {REASON_MAX_CHARS} characters.")
             return
 
-        # Manual request with `raise_for_status` as False because we want the actual response
-        session = self.bot.api_client.session
-        url = self.bot.api_client._url_for('bot/nominations')
-        kwargs = {
-            'json': {
-                'actor': ctx.author.id,
-                'reason': reason,
-                'user': user.id
-            },
-            'raise_for_status': False,
-        }
-        async with session.post(url, **kwargs) as resp:
-            response_data = await resp.json()
-
-            if resp.status == 400:
-                if response_data.get('user', False):
+        try:
+            await self.api.post_nomination(user.id, ctx.author.id, reason)
+        except ResponseCodeError as e:
+            match (e.status, e.response_json):
+                case (400, {"user": _}):
                     await ctx.send(f":x: {user.mention} can't be found in the database tables.")
-                elif response_data.get('actor', False):
+                    return
+                case (400, {"actor": _}):
                     await ctx.send(f":x: You have already nominated {user.mention}.")
-
-                return
-            else:
-                resp.raise_for_status()
-
-        self.cache[user.id] = response_data
+                    return
+            raise
```

Comment:
Should there be an explicit exception raised instead of the plain `raise` usages here?
Currently if this does raise, it would trace the following:
```py
RuntimeError: No active exception to reraise
```
which might cause confusion later. And since we're matching a specific 400 code, it's pretty trivial for these matches to be non-exhaustive and hit the raise.

Reply:
It's still in the `except` block so should just reraise that exception, right?

Reply:
It does, I got confused 👍